portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,32 +15,62 @@
 package com.google.sps.servlets;
 
 import java.util.ArrayList;
+import java.util.List;
 import com.google.sps.data.DataComment;
 import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+
+/** Servlet responsible for creating new comments. */
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
 
-  private DataComment data;
+  private ArrayList<String> comments = new ArrayList<>(); 
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getParameter("new-comment");
+    comments.add(text);
+
+    // Store comments in datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", text);
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/index.html");
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    DataComment object = new DataComment("mahima", "hello");
-    String json = convertToJson(object);
+    Query query = new Query("Comment").addSort("text", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
You already have a datastore as an instance variable, so you don't need to get another one here

Reply:
removed!